### PR TITLE
Add optional Crawlera support

### PR DESCRIPTION
**Pull request by vshlapakov**
Supersedes #4 (not I nor Rafael has write access to the repository, so we have to use forks, and while Rafael is on vacation, I need to finish his work on the integration).

As discussed, Crawlera isn't used for AutoExtract requests itself, but for everything else. I also added sessions support to maintain same session during crawling. Looking forward to get any feedback on the changes.

### CODE REVIEW

File: autoextract_spiders/sessions.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+def update_redirect_middleware(settings):
+    if settings.getbool('CRAWLERA_ENABLED'):
+        redirect_mware = 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware'
+        pos = settings.get('DOWNLOADER_MIDDLEWARES_BASE').pop(redirect_mware)
+        DW_MIDDLEWARES = settings.get('DOWNLOADER_MIDDLEWARES')
+        DW_MIDDLEWARES['crawlera_session.CrawleraSessionRedirectMiddleware'] = pos
+
+
+class RequestSession(_Session):
```
Comment (vshlapakov) [inline autoextract_spiders/sessions.py:13]: The custom class is needed mainly to make the integration switchable, as the current implementation fails when some requests don't have expected headers/metadata. We could probably migrate the logic to crawlera-sessions repo, and drop it from here: it doesn't make a lot of sense to setup sessions if Crawlera is disabled.

File: autoextract_spiders/sessions.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+from crawlera_session import RequestSession as _Session
+
+
+def update_redirect_middleware(settings):
+    if settings.getbool('CRAWLERA_ENABLED'):
+        redirect_mware = 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware'
```
Comment (vshlapakov) [inline autoextract_spiders/sessions.py:7]: Just a convenient helper to move another piece of crawlera-only logic to the single place

**Comment by kmike** [comment]
Thanks @vshlapakov and @rafaelcapucho! I haven't checked in detail how crawlera sessions work here, but the code looks good overall, merging it 👍

**Comment by kmike** [comment]
One more thing: this needs to be described in README.

**Pull request merged**